### PR TITLE
hwmv2: Port all arc boards

### PR DESCRIPTION
**Pull request by nordicjm**
tests: Update board names for hwmv2
    
    Updates tests that use board names which have changed with boards v2
    

    boards: arc: hsdk4xd: Convert to v2
    
    Converts the board to hwmv2

    soc: snps_arc_hsdk4xd: Port to HWMv2
    
    Ports the snps_arc_hsdk4xd SoC configuration to hardware model
    version 2

    boards: arc: iotdk: Convert to v2
    
    Converts the board to hwmv2

    soc: snps_arc_iot: Port to HWMv2
    
    Ports the snps_arc_iot SoC configuration to hardware model version 2

    boards: arc: emsdp: Convert to v2
    
    Converts the board to hwmv2

    soc: snps_emsdp: Port to HWMv2
    
    Ports the snps_emsdp SoC configuration to hardware model version 2

    boards: arc: em_starterkit: Convert to v2
    
    Converts the board to hwmv2, documentation for this board needs to
    be vastly updated
    

    soc: snps_emsk: Port to HWMv2
    
    Ports the snps_emsk SoC configuration to hardware model version 2
    

    boards: arc: hsdk: Convert to v2
    
    Converts the board to hwmv2
    

    soc: snps_arc_hsdk: Port to HWMv2
    
    Ports the snps_arc_hsdk SoC configuration to hardware model
    version 2

    boards: arc: nsim: Convert to v2
    
    Converts the board to hwmv2

    soc: snps_nsim: Port to HWMv2
    
    Ports the snps_nsim SoC configuration to hardware model version 2

    boards: arc: qemu_arc: Convert to v2
    
    Converts the board to hwmv2
    

    soc: snps_qemu: Port to HWMv2
    
    Ports the snps_qemu SoC configuration to hardware model version 2

### CODE REVIEW

File: boards/v2/qemu/qemu_arc/board.yml
Comment (gmarull) [inline boards/v2/qemu/qemu_arc/board.yml:3]: since the board is under `qemu`, having vendor: Synopsys is a bit confusing. What is the policy with qemu boards?

Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/board.yml]: Fixed

File: boards/v2/qemu/qemu_arc/Kconfig.qemu_arc
Comment (tejlmand) [inline boards/v2/qemu/qemu_arc/Kconfig.qemu_arc:6]: bool not needed.

And as all board identifier are doing: `select BOARD_QEMU_ARC`, then let's move that up, and remove redundant code.
```suggestion
	select BOARD_QEMU_ARC
```

File: boards/v2/qemu/qemu_arc/Kconfig.qemu_arc
Comment (tejlmand) [inline boards/v2/qemu/qemu_arc/Kconfig.qemu_arc:26]: If moving the `select` to board level, then all those lines can be removed.
```suggestion
```

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (tejlmand) [inline boards/v2/synopsys/nsim/Kconfig.nsim:7]:
```suggestion
```

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (tejlmand) [inline boards/v2/synopsys/nsim/Kconfig.nsim:15]:
```suggestion
```

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (tejlmand) [inline boards/v2/synopsys/nsim/Kconfig.nsim:19]:
```suggestion
```

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (tejlmand) [inline boards/v2/synopsys/nsim/Kconfig.nsim:23]:
```suggestion
```

please cleanup following lines, even though not commented directly.

File: boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit
Comment (tejlmand) [inline boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit:17]:
```suggestion
```

File: boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit
Comment (tejlmand) [inline boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit:21]:
```suggestion
```

File: boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit
Comment (tejlmand) [inline boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit:25]:
```suggestion
```

File: boards/v2/synopsys/emsdp/Kconfig.emsdp
Comment (tejlmand) [inline boards/v2/synopsys/emsdp/Kconfig.emsdp:17]: Please remove redundant `bool` in this file.

```suggestion
```

File: boards/v2/synopsys/iotdk/board.yml
Comment (tejlmand) [inline boards/v2/synopsys/iotdk/board.yml:5]: newline missing.

File: boards/v2/synopsys/hsdk4xd/board.yml
Comment (tejlmand) [inline boards/v2/synopsys/hsdk4xd/board.yml:5]: new line missing.

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: boards/v2/qemu/qemu_arc/board.yml
Comment (ruuddw) [reply boards/v2/qemu/qemu_arc/board.yml]: fyi, we had this discussion in the past. The ARC qemu is done by Synopsys, and I think not yet (at least then) upstream. That was the reason for keeping Synopsys as vendor.
But, I'm fine with moving under qemu.

File: boards/v2/qemu/qemu_arc/Kconfig
Comment (ruuddw) [inline boards/v2/qemu/qemu_arc/Kconfig]: I'd prefer sticking to "em" for this one, instead of "em4". This is a 'generic' board for the family, e.g. covering em6 as well. And, there are few more places (e.g. support/mdb_em.args) that have the same name.
Let's not change names as part of this restructuring. If for some reason it needs to be changed, then have a separate PR for that.

File: boards/v2/qemu/qemu_arc/Kconfig
Comment (ruuddw) [inline boards/v2/qemu/qemu_arc/Kconfig]: same for this one, let's keep this as generic 'hs'.

File: boards/v2/qemu/qemu_arc/Kconfig.qemu_arc
Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig.qemu_arc]: Fixed

File: boards/v2/qemu/qemu_arc/Kconfig.qemu_arc
Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig.qemu_arc]: Fixed

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (nordicjm) [reply boards/v2/synopsys/nsim/Kconfig.nsim]: Fixed

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (nordicjm) [reply boards/v2/synopsys/nsim/Kconfig.nsim]: Fixed

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (nordicjm) [reply boards/v2/synopsys/nsim/Kconfig.nsim]: Fixed

File: boards/v2/synopsys/nsim/Kconfig.nsim
Comment (nordicjm) [reply boards/v2/synopsys/nsim/Kconfig.nsim]: Fixed

File: boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit
Comment (nordicjm) [reply boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit]: Fixed

File: boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit
Comment (nordicjm) [reply boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit]: Fixed

File: boards/v2/synopsys/emsdp/Kconfig.emsdp
Comment (nordicjm) [reply boards/v2/synopsys/emsdp/Kconfig.emsdp]: Fixed

File: boards/v2/synopsys/iotdk/board.yml
Comment (nordicjm) [reply boards/v2/synopsys/iotdk/board.yml]: Fixed

File: boards/v2/synopsys/hsdk4xd/board.yml
Comment (nordicjm) [reply boards/v2/synopsys/hsdk4xd/board.yml]: Fixed

**Review by ruuddw — CHANGES_REQUESTED** [review]

File: boards/v2/qemu/qemu_arc/Kconfig
Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig]: This is the CPU, the CPU being em4, it cannot be em because there is no such soc

File: boards/v2/qemu/qemu_arc/Kconfig
Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig]: As above

**Comment by tejlmand** [comment]
please rebase

File: boards/v2/qemu/qemu_arc/Kconfig
Comment (ruuddw) [reply boards/v2/qemu/qemu_arc/Kconfig]: The original SoC was also called EM, not sure I understand the issue with sticking with EM and HS instead of renaming all of these (globally: soc/board/docs).

Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig]: Why were two soc added with invalid names originally? em and hs are not SoCs, they are soc families, em4 and hs3x are the soc names according to https://www.synopsys.com/designware-ip/processor-solutions/arc-classic-processors/arc-hs-family.html and https://www.synopsys.com/designware-ip/processor-solutions/arc-classic-processors/arc-em-family.html

Comment (ruuddw) [reply boards/v2/qemu/qemu_arc/Kconfig]: ARC is highly configurable, instead of adding a SoC for all possible configurations we made an abstraction.
nsim_em e.g. covers both EM4 (no caches) and EM6 (with caches).
em4 and hs3x are not SoC names. These are product names for Synopsys ARC CPU IP cores (which can be configured and integrated on a SoC).

Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig]: With hwmv2, there is not the level of configurability as was present with hwmv1, so for example with hwmv1 you could configure for an stm32 board which has a soc, but you could then change the soc part number, in hwmv2 this is no longer possible, you configure and build for a specific soc, if you want a different part number, you need to alter the board files to specify that soc (if it isn't already one of the possible soc options) but you cannot change the configuration once configured. So with the em and hs socs, is that what you mean by configurability or is it something else?

Comment (ruuddw) [reply boards/v2/qemu/qemu_arc/Kconfig]: No, I mean silicon IP core configurability. That is several steps before board/soc configurability.
Let's separate hwmv2 porting and other possible improvements like changing names. My goal is to minimize changes for existing users. If you feel strongly about the names, then let's have a separate issue/PR/discussion about that.

**Review by tejlmand — APPROVED** [review]

File: boards/v2/qemu/qemu_arc/Kconfig
Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig]: You can't have this both ways, the soc is hs3x and it was wrong to have hs in the original board then have hs3x_xip (which is a variant), you can either have both called hs (which in my opinion is wrong because there is no hs soc) or both named hs3x.

Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig]: The options here are as follows:

- The naming in this PR is picked, em4 and hs3x, to build for the hs3x you would use `qemu_arc/qemu_hs3x` and for XIP variant `qemu_arc/qemu_hs3x/xip`
- The naming in this PR is changed so that em4 falls back to em and hs3x falls back to hs, this means the hs3x_xip name will vanish and be hs
- The existing names from hwmv1 are used, this means the whole hs3x soc has to be duplicated as hs and hs3x, this then means hs and hs3x will be different boards in qemu, it means you will have `qemu_arc/qemu_hs`, `qemu_arc/qemu_hs3x` (which will fail to configure and throw an error, it will still appear in the list of supported board names) and `qemu_arc/qemu_hs3x/xip` which will be the XIP version

Comment (nordicjm) [reply boards/v2/qemu/qemu_arc/Kconfig]: I got mixed up between different socs it seems, it has been addressed

**Comment by nordicjm** [comment]
@ruuddw please review

**Comment by ruuddw** [comment]
I still see the renamings for EM and HS and still fail to understand why these are needed. Please use "EM" instead of "EM4" and "HS" instead of "HS3X". The XIP variant can be qemu_arc/qemu_hs/xip

Furthermore, do I need to do anything special to build? I tried some v2 boards, but get various errors:
warning: SPARC_CASA (defined at soc\sparc\Kconfig:4) defined without a type
warning: attempt to assign the value 'n' to the undefined symbol ARC_HAS_STACK_CHECKING
warning: attempt to assign the value 'y' to the undefined symbol ARC_MPU_ENABLE

**Comment by nordicjm** [comment]
> I still see the renamings for EM and HS and still fail to understand why these are needed. Please use "EM" instead of "EM4" and "HS" instead of "HS3X". The XIP variant can be qemu_arc/qemu_hs/xip
> 
> Furthermore, do I need to do anything special to build? I tried some v2 boards, but get various errors: warning: SPARC_CASA (defined at soc\sparc\Kconfig:4) defined without a type warning: attempt to assign the value 'n' to the undefined symbol ARC_HAS_STACK_CHECKING warning: attempt to assign the value 'y' to the undefined symbol ARC_MPU_ENABLE

qemu_arc was renamed, can you point out exactly what boards have been renamed? That is a path issue in windows, if you rebase on `collab-hwm` it will be resolved

**Review by gmarull — DISMISSED** [review]
update docs/maintainers as needed

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em7d.dts
Comment (ruuddw) [inline boards/synopsys/em_starterkit/em_starterkit_emsk_em7d.dts:10]: why "-sec"?

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em9d.dts
Comment (ruuddw) [inline boards/synopsys/em_starterkit/em_starterkit_emsk_em9d.dts:11]: why  "_r22" instead of "_r23"?

**Review by ruuddw — CHANGES_REQUESTED** [review]
Not fully finished my review, but it looks you're updating in parallel so I'll check again later.
Please let me know when to review again.

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em11d.dts
Comment (nordicjm) [reply boards/synopsys/em_starterkit/em_starterkit_emsk_em11d.dts:11]: The original board files were wrong, the documentation states that em11d is supported on r2.3 and r2.2 but actually did not allow selecting r2.2

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em7d.dts
Comment (nordicjm) [reply boards/synopsys/em_starterkit/em_starterkit_emsk_em7d.dts:10]: because that is what the original board dts has?

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em9d.dts
Comment (nordicjm) [reply boards/synopsys/em_starterkit/em_starterkit_emsk_em9d.dts:11]: > The original board files were wrong, the documentation states that em11d is supported on r2.3 and r2.2 but actually did not allow selecting r2.2

With these changes, the boards can correctly be changed between r2.2 and r2.3

File: boards/qemu/arc/doc/index.rst
Comment (nordicjm) [reply boards/qemu/arc/doc/index.rst]: Fixed

File: boards/qemu/arc/doc/index.rst
Comment (nordicjm) [reply boards/qemu/arc/doc/index.rst]: Fixed

File: boards/qemu/arc/doc/index.rst
Comment (nordicjm) [reply boards/qemu/arc/doc/index.rst]: Fixed

File: boards/synopsys/em_starterkit/doc/index.rst
Comment (nordicjm) [reply boards/synopsys/em_starterkit/doc/index.rst]: Fixed

File: boards/synopsys/em_starterkit/doc/index.rst
Comment (nordicjm) [reply boards/synopsys/em_starterkit/doc/index.rst]: Fixed

File: boards/synopsys/em_starterkit/doc/index.rst
Comment (nordicjm) [reply boards/synopsys/em_starterkit/doc/index.rst]: Fixed

File: boards/synopsys/em_starterkit/doc/index.rst
Comment (nordicjm) [reply boards/synopsys/em_starterkit/doc/index.rst]: Fixed

**Comment by nordicjm** [comment]
@ruuddw ready for review

File: soc/synopsys/emsdp/Kconfig.soc
Comment (ruuddw) [inline soc/synopsys/emsdp/Kconfig.soc:55]: This looks like a typo there is no EM6D only EM6.

**Review by ruuddw — CHANGES_REQUESTED** [review]

File: boards/synopsys/index.rst
Comment (gmarull) [reply boards/synopsys/index.rst:3]: -1, all pages follow vendor name only, let’s be consistent please

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em11d.dts
Comment (nordicjm) [reply boards/synopsys/em_starterkit/em_starterkit_emsk_em11d.dts]: Removed

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em7d.dts
Comment (nordicjm) [reply boards/synopsys/em_starterkit/em_starterkit_emsk_em7d.dts]: Fixed

File: boards/synopsys/em_starterkit/em_starterkit_emsk_em9d.dts
Comment (nordicjm) [reply boards/synopsys/em_starterkit/em_starterkit_emsk_em9d.dts]: Removed

File: soc/synopsys/emsdp/Kconfig.soc
Comment (nordicjm) [reply soc/synopsys/emsdp/Kconfig.soc]: Fixed

File: boards/synopsys/nsim/board.yml
Comment (nordicjm) [reply boards/synopsys/nsim/board.yml:6]: This is because you insist on em and em7d being different socs when they're really the same soc of a different revision

**Review by ruuddw — APPROVED** [review]
Thanks for the work! This looks ok now, we can do more testing once this gets merged and we have updated out internal regression environment for hwmv2

**Pull request merged**
